routes/routes_impl.py

```diff
@@ -615,6 +622,45 @@ def bathymetry_impl(projection: str, zoom: int, x: int, y: int):
         img = plotting.tile.bathymetry(projection, x, y, zoom, {})
         return _cache_and_send_img(img, f)
 
+def mbt_impl(projection: str, tiletype: str, zoom: int, x: int, y: int):
+  """
+       Serves mbt files
+  """
+  cache_dir = current_app.config['CACHE_DIR']
+  shape_file_dir = current_app.config['SHAPE_FILE_DIR']
+  directory = str(os.path.join(cache_dir, request.path[1:]))
```

Comment:
Is `directory` actually a directory path, or does it end with some file extension? I'm asking because on line 660 and 662 `directory` is being used as a filepath.

Reply:
Sorry about this. I've changed some of the variable names just now so it makes a bit more sense:
`requestf` is the mbt file (no extension) that is being requested by the user
`basedir` is the directory in which that file is stored, or where it is meant to be stored